c2corg_ui/views/route.py

```diff
@@ -10,28 +11,39 @@ class Route(Document):
     @view_config(route_name='routes_index',
                  renderer='c2corg_ui:templates/route/index.html')
     def index(self):
-        url = '%s/%s' % (self.settings['api_url'], self._API_ROUTE)
-        resp, content = self._call_api(url)
-        # TODO: error handling (not found, etc.)
-        return {
-            'debug': 'debug' in self.request.params,
-            'routes': content if resp.status == 200 else {}
-        }
+        self.template_input.update({
+            'routes': self._get_documents()
+        })
+        return self.template_input
 
     @view_config(route_name='routes_view',
                  renderer='c2corg_ui:templates/route/view.html')
     def view(self):
         id, culture = self._validate_id_culture()
-        # TODO: get only the current culture
-        url = '%s/%s/%d' % (
-            self.settings['api_url'],
-            self._API_ROUTE,
-            int(self.request.matchdict['id'])
-        )
-        resp, content = self._call_api(url)
-        # TODO: error handling
-        return {
-            'debug': 'debug' in self.request.params,
+        route, locale, other_cultures = self._get_document(id, culture)
+        self.template_input.update({
             'culture': culture,
-            'route': content if resp.status == 200 else {}
-        }
+            'route': route,
+            'locale': locale,
+            'other_cultures': other_cultures
+        })
+        return self.template_input
+
+    @view_config(route_name='routes_add',
+                 renderer='c2corg_ui:templates/route/edit.html')
+    @view_config(route_name='routes_edit',
+                 renderer='c2corg_ui:templates/route/edit.html')
+    def edit(self):
+        try:
+            id, culture = self._validate_id_culture()
+        except Exception:
+            id = None
```

Comment:
When will you get in here?
- When you try to edit a document with an invalid id and culture? I think this case should be handled with a generic error handler?
- When you try to add a new document? There are cleaner ways to check if you want to add a new document.